clients/RemoteDisplay/transport_plugin_udp.c

```diff
@@ -243,15 +252,19 @@ send_packet(uint8_t *payload, size_t size, uint32_t timestamp,
 	}
 
 	for(i = 0; i < private_data->num_addr; i++) {
-		int rval = sendto(private_data->socket[i].sockDesc,
-				rtpBase8,
-				size + RTP_HEADER_SIZE,
-				0,
-				(struct sockaddr *) &private_data->socket[i].sockAddr,
-				sizeof(private_data->socket[i].sockAddr));
-
-		if (rval <= 0) {
-			fprintf(stderr, "Send failed with %m\n");
+		if (TRUE == private_data->socket[i].available) {
+			int rval = sendto(private_data->socket[i].sockDesc,
+					rtpBase8,
+					size + RTP_HEADER_SIZE,
+					0,
+					(struct sockaddr *) &private_data->socket[i].sockAddr,
+					sizeof(private_data->socket[i].sockAddr));
+
+			if (rval <= 0) {
```

Comment:
Should we stop sending as soon as we have the first error Or should we have a threshold after which we stop sending?
Also, should we stop on any kind of error or look for rval to be a certain type of error (like the other side is not available)?

Reply:
An encoded frame is divided into tens of UDP packets.
There is no guarantee sending more packets after the 1st error is helpful to send whole frame data.
Thus it should be better to stop sending and restart to send from new frame.

About kind of error, I will check the error types more and handle it accordingly.

Reply:
If a receiver machine is rebooted, will this resume sending to that when it comes back up? Particularly, while sending to 2 machines and if one is rebooted, will not this set available to false and never try after?

Reply:
Hi Gowtham,

Thank for your input.
One encoded frame is split into tens of UDP packets and my code stops sending rest of UDP packet for a frame if a receiver's network is unreachable by setting the available variable to false.
This available variable for all receivers is set as true at the end of send frame() to try sending a new frame to receivers.  
In the case of being rebooted, a sender stops sending frames till network on the rebooted receiver is fully on and then sending frames again to it.